probe-rs/src/bin/probe-rs/cmd/dap_server/debug_adapter/dap/repl_commands.rs

```diff
@@ -361,4 +362,55 @@ pub(crate) static REPL_COMMANDS: &[ReplCommand<ReplHandler>] = &[
             memory_read(input_address, gdb_nuf, target_core)
         },
     },
+    ReplCommand {
+        command: "dump",
+        help_text: "Create a core dump at a target location",
+        sub_commands: None,
+        args: Some(&[
+            ReplCommandArgs::Optional("memory start address"),
+            ReplCommandArgs::Optional("memory size in bytes"),
+        ]),
+        handler: |target_core, command_arguments, _request_arguments| {
+            let input_arguments = command_arguments.split_whitespace();
+            let location = Path::new("./coredump");
+
+            let ranges = input_arguments
+                .collect_vec()
+                .chunks(2)
+                .map(|c| {
+                    let start = if let Some(start) = c.get(0) {
+                        parse_int::parse::<u64>(start)
+                            .map_err(|e| DebuggerError::UserMessage(e.to_string()))?
+                    } else {
+                        return Err(DebuggerError::UserMessage(
+                            "A stack start is required.".to_string(),
+                        ));
+                    };
+
+                    let size = if let Some(size) = c.get(0) {
+                        parse_int::parse::<u64>(size)
+                            .map_err(|e| DebuggerError::UserMessage(e.to_string()))?
+                    } else {
+                        return Err(DebuggerError::UserMessage(
+                            "A memory size is required.".to_string(),
+                        ));
+                    };
+
+                    Ok(start..start + size)
+                })
+                .collect::<Result<Vec<_>, _>>()?;
+
+            target_core.core.dump(ranges)?.store(location)?;
+
+            Ok(Response {
+                command: "dump".to_string(),
+                success: true,
+                message: None,
+                type_: "response".to_string(),
+                request_seq: 0,
+                seq: 0,
```

Comment:
Not sure what proper values would be, I copy pasted this.

Reply:
Check out the doc comment on line 15-22 onwards. These fields are ignored, and the actual DAP Request / Response fields associated with the original `Evaluate` request are maintained at the call location.  In hindsight, we could probably use a custom struct here to avoid confusion in future. 

@Yatekii, if you want to add a TODO: in the doc comment on line 15, I will clean this up at a future date.

Reply:
Hihihi this is your old C-you that coded this :D